app/uk/gov/hmrc/pdfgenerator/service/PdfGeneratorService.scala

```diff
@@ -89,31 +94,76 @@ class PdfGeneratorService @Inject()(configuration: Configuration, resourceHelper
       ADOBE_COLOR_PROFILE, ADOBE_COLOR_PROFILE_FULL_PATH)
   }
 
-  def generateCompliantPdfA(html: String): Try[File] = {
-
+  def generatePdf(html: String, forcePdfA: Boolean): Try[File] = {
     logConfig()
-
     val inputFileName: String = UUID.randomUUID.toString + ".pdf"
     val outputFileName: String = UUID.randomUUID.toString + ".pdf"
-    Logger.trace(s"generateCompliantPdfA from ${html}")
-    Logger.info(s"generateCompliantPdfA inputFileName: ${inputFileName} outputFileName: ${outputFileName}")
+    Logger.trace(s"generatePdf from $html")
+    val linksDisabled = if(forcePdfA) true else getLinksDisabled(html)
+
+    try {
+
+      val triedFile = generatePdfFromHtml(html, BASE_DIR + inputFileName, linksDisabled)
+      if(linksDisabled){
+        triedFile.flatMap(_ => convertToPdfA(getBaseDir + inputFileName, getBaseDir + outputFileName))
+      } else {
+          Logger.warn("*** Generated PDF will not be PDF/A compliant as it contains valid gov.uk links ***")
+          triedFile
+        }
+      } finally { if(!linksDisabled) deleteFile(BASE_DIR + inputFileName) }
+  }
 
-    def cleanUpInputFile = {
-      val inputFile = new File(BASE_DIR + inputFileName)
-      if (inputFile.exists()) {
-        inputFile.delete()
-      }
+  private def deleteFile(fileName: String) = {
+    val file = new File(BASE_DIR + fileName)
+    if (file.exists()) {
+      file.delete()
     }
+  }
 
-    val triedFile = generatePdfFromHtml(html, BASE_DIR + inputFileName)
-      .flatMap(_ => convertToPdfA(getBaseDir + inputFileName, getBaseDir + outputFileName))
+  /**
+    * Returns false if hyperlinks in source HTML are valid and true if links are invalid
+    * @param html source HTML
+    * @return true or false
+    */
+  def getLinksDisabled(html: String): Boolean = {
+    val startTime = System.currentTimeMillis()
+    val links = extractLinksFromHtml(html)
+    Logger.trace("Checking document for links took " + (System.currentTimeMillis() - startTime) + " milliseconds")
+    if(onlyContainsValidLinks(links)) false else true
+  }
 
-    cleanUpInputFile
-    triedFile
+  private def extractLinksFromHtml(html: String): List[String] = {
+    val doc: Document = Jsoup.parse(html)
+    Option(doc.getElementsByTag("a")) match {
```

Comment:
Does this need to be wrapped in Option? getElementsByTag return empty list if it doesn't match